buildSrc/subprojects/jvm/src/main/kotlin/gradlebuild.unittest-and-compile.gradle.kts

```diff
@@ -35,6 +35,8 @@ plugins {
 }
 
 extensions.create<UnitTestAndCompileExtension>("gradlebuildJava", java)
+apply(from = "${rootProject.projectDir}/gradle/shared-with-buildSrc/fail-on-empty-source-dirs.gradle.kts")
```

Comment:
When this part is executed during compilation of the script, it fails, because `rootProject.projectDir` is some artificial directory.
You could work around this by checking if the files exists before applying it.

Reply:
Would that mean that in some cases the workaround is not applied?

Reply:
When the script is compiled, an artificial build is configured to get all the information for compilation (e.g. which accessors need to be generated). In that case, the file can't be found. If you add the check the _fail-on-empty_ script plugin is then ignored.

When the build runs for real, the script plugin will be found and used as expected.

Reply:
Ah interesting! Good to know.